specs/ContextMenuRequested.md

```diff
@@ -0,0 +1,649 @@
+
+        /// Context menu item for "Copy image" action.
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_COPY_IMAGE,
+
+        /// Context menu item for "Copy image link" action.
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_COPY_IMAGE_LINK,
+
+        /// Context menu item for "Save media as" action.
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_SAVE_MEDIA_AS,
+
+        /// Context menu item for other commands not corresponding with any of the enums above.
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_OTHER,
+    } COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR;
+    
+    /// Indicates the type of selected context.
+    [v1_enum]
+    typedef enum COREWEBVIEW2_CONTEXT_TYPE
+    {
+        /// Indicates that the context menu was created for the page.
+        COREWEBVIEW2_CONTEXT_TYPE_PAGE,
+
+        /// Indicates that the context menu was created for a frame.
+        COREWEBVIEW2_CONTEXT_TYPE_FRAME,
+
+        /// Indicates that the context menu was created for a selection.
+        COREWEBVIEW2_CONTEXT_TYPE_SELECTION,
+
+        /// Indicates that the context menu was created for a link.
+        COREWEBVIEW2_CONTEXT_TYPE_LINK,
+
+        /// Indicates that the context menu was created for an editable component.
+        COREWEBVIEW2_CONTEXT_TYPE_EDITABLE,
+        
+        /// Indicates that the context menu was created for audio.
+        COREWEBVIEW2_CONTEXT_TYPE_AUDIO,
+
+        /// Indicates that the context menu was created for an image.
+        COREWEBVIEW2_CONTEXT_TYPE_IMAGE,
+        
+        /// Indicates that the context menu was created for a video.
+        COREWEBVIEW2_CONTEXT_TYPE_VIDEO,
+    } COREWEBVIEW2_CONTEXT_TYPE;
+
+    /// Indicates the menu item type.
+    [v1_enum]
+    typedef enum COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND
+    {
+        /// A normal menu item kind, can be enabled or not.
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND_NORMAL,
+
+        /// A checkbox. Can be checked or enabled.
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND_CHECKBOX,
+
+        /// A radio button type, can be checked or enabled.
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND_RADIO,
+
+        /// A separator type, used to visually differentiate sections of context menu items.
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND_SEPARATOR,
+
+        /// A submenu type, will contain a list of its children menu items.
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND_SUBMENU,
+    } COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND;
+
+    /// Represents a context menu item of a context menu displayed by WebView.
+    [uuid(7aed49e3-a93f-497a-811c-749c6b6b6c65), object, pointer_default(unique)]
+    interface ICoreWebView2ContextMenuItem : IUnknown
+    {
+        /// Get the label displayed for the ContextMenuItem, will already have been translated by the WebView.
+        [propget] HRESULT Label([out, retval] LPWSTR* value);
+
+        /// Get the name for the ContextMenuItem, will be an unlocalized string to allow for 
```

Comment:
The name would most likely be set for all of them but will only be in use for the extension items and other items since those are the two items that can have multiple items that fall in those enum categories. I will specify that in the description.